src/util/deleteThought.ts

```diff
@@ -112,14 +111,16 @@ export const deleteThought = () => {
     }
   }
 
-  setCursorOrBack(...prev ? [contextOf(path).concat(prev), { offset: prev.value.length }] :
+
+  if (prev) {
+    return setCursorOrBack(contextOf(path).concat(prev!), { offset: prev.value.length })
+  }
   // Case II: set cursor on next thought
-    next() ? [showContexts
-      ? contextOf(path).concat({ value: head(next().context), rank: next().rank })
-      : contextOf(path).concat(next()), { offset: 0 }] :
+  if (next()) {
+    return showContexts ? setCursorOrBack(contextOf(path).concat({ value: head(next().context), rank: next().rank })) 
     // Case III: delete last thought in context; set cursor on context
-    thoughts.length > 1 ? [rootedContextOf(path), { offset: head(context).length }]
-    // Case IV: delete very last thought; remove cursor
-    : [null]
-  )
+    : setCursorOrBack(contextOf(path).concat(next()), { offset: 0 })
+  }
+  // Case IV: delete very last thought; remove cursor
+  return thoughts.length>1 ?setCursorOrBack(rootedContextOf(path), { offset: head(context).length }) : setCursorOrBack([])
 }
```

Comment:
Do not use multiple return statements, as it breaks the invariant that the shape of the code should reflect the control flow. A single return statement with multiple cases captures the logic with less redundancy.